nim_status/client.nim

```diff
@@ -90,58 +153,228 @@ proc storeDerivedAccountsInDb(self: StatusObject, id: UUID, keyUid: string,
       chat: true.some,
       `type`: some($AccountType.Seed),
       storage: string.none,
-      path: paths[2].some,
+      path: PATH_WHISPER.some,
       publicKey: whisperAccountPubKeyResult.get.some,
       name: pubAccount.name.some,
       color: "#4360df".some
     )
-  self.userDb.createAccount(defaultWalletAccount)
-  self.userDb.createAccount(whisperAccount)
 
-  PublicAccountResult.ok(pubAccount)
+  try:
+    # We need an inited user db to create accounts, which requires a login.
+    # First, record if we are currently logged in, and then init the user db
+    # if not. After we know the db has been inited, create the needed accounts.
+    # Once finished, close the db if we were originally logged out.
+    let wasLoggedIn = self.isLoggedIn
+    if not wasLoggedIn:
+      self.initUserDb(keyUid, password)
+
+    self.userDb.createAccount(defaultWalletAccount)
+    self.userDb.createAccount(whisperAccount)
+
+    if not wasLoggedIn:
+      self.closeUserDb()
 
-proc createAccount*(self: StatusObject,
-  mnemonicPhraseLength, n: int, bip39Passphrase, password: string,
-  paths: seq[KeyPath], dir: string): PublicAccountResult =
+    return PublicAccountResult.ok(pubAccount)
+  except Exception as e:
+    return PublicAccountResult.err e.msg
+
+proc addWalletAccount*(self: StatusObject, name, password,
+  #[privateKey: Option[string],]# dir: string): WalletAccountResult =
```

Comment:
Quite ignorant about it - does this mean we might possibly support specifying private keys for wallet account creation?

Reply:
Yes, it is something I am working on now. I have decided against including the private key as a parameter and will create a new proc for it.